internal/pkg/services/secrets-manager/utils/utils_test.go

```diff
@@ -79,3 +90,54 @@ func TestGetInstanceName(t *testing.T) {
 		})
 	}
 }
+
+func TestGetUserDetails(t *testing.T) {
+	tests := []struct {
+		description    string
+		getUserFails   bool
+		GetUserResp    *secretsmanager.User
+		isValid        bool
+		expectedOutput [2]string
+	}{
+		{
+			description: "base",
+			GetUserResp: &secretsmanager.User{
+				Username:    utils.Ptr(testUserName),
+				Description: utils.Ptr(testDescription),
+			},
+			isValid:        true,
+			expectedOutput: [2]string{testUserName, testDescription},
+		},
+		{
+			description:  "get instance fails",
+			getUserFails: true,
+			isValid:      false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.description, func(t *testing.T) {
+			client := &secretsManagerClientMocked{
+				getUserFails: tt.getUserFails,
+				getUserResp:  tt.GetUserResp,
+			}
+
+			username, description, err := GetUserDetails(context.Background(), client, testProjectId, testInstanceId, testUserId)
+
+			output := [2]string{username, description}
```

Comment:
Maybe we don't need the extra array, we could just define an `expectedUsername` and `expectedDescription` and compare both strings